holidays/countries/cambodia.py

```diff
@@ -0,0 +1,342 @@
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/dr-prodigy/python-holidays
+#  License: MIT (see LICENSE file)
+
+from datetime import timedelta as td
+from gettext import gettext as tr
+
+from holidays.calendars import _ThaiLunisolar, KHMER_CALENDAR
+from holidays.constants import JAN, APR, MAY, JUN, AUG, SEP, OCT, NOV, DEC
+from holidays.holiday_base import HolidayBase
+from holidays.holiday_groups import InternationalHolidays
+
+
+class Cambodia(HolidayBase, InternationalHolidays):
+    """
+    A subclass of :py:class:`HolidayBase` representing public holidays
+    in Cambodia.
+
+
+    References:
+
+    - Based on: https://www.nbc.gov.kh/english/news_and_events/official_holiday.php
+                https://www.nbc.gov.kh/news_and_events/official_holiday.php
+                https://en.wikipedia.org/wiki/Public_holidays_in_Cambodia
+
+    - Checked with: https://asean.org/wp-content/uploads/2021/12/ASEAN-National-Holidays-2022.pdf  # noqa: E501
+                    https://asean.org/wp-content/uploads/2022/12/ASEAN-Public-Holidays-2023.pdf  # noqa: E501
+                    https://www.timeanddate.com/holidays/cambodia/
+
+    Limitations:
+
+    - Cambodian holidays only works from 1993 onwards.
+
+    - Exact Public Holidays as per Cambodia's Official Gazette are only
+      available from 2015 onwards.
+
+    - Cambodian Lunar Calendar Holidays only work from 1941 (B.E. 2485) onwards
+      until 2057 (B.E. 2601) as we only have Thai year-type data for
+      cross-checking until then for Cambodian Buddhist Holidays.
+
+
+    Country created by: `PPsyrius <https://github.com/PPsyrius>`__
+
+    Country maintained by: `PPsyrius <https://github.com/PPsyrius>`__
+    """
+
+    country = "KH"
+    default_language = "km"
+
+    # Special Cases.
+
+    sangkranta_in_lieu_covid = tr(
+        # Khmer New Year's Replacement Holiday
+        "ថ្ងៃឈប់សម្រាកសងជំនួសឲ្យពិធីបុណ្យចូលឆ្នាំថ្មីប្រពៃណីជាតិ"
+    )
+    # Special Public Holiday
+    special_in_lieu_holidays = tr("ថ្ងៃឈប់សម្រាកសងជំនួស")
+
+    special_holidays = {
+        2016: (
+            (MAY, 2, special_in_lieu_holidays),
+            (MAY, 16, special_in_lieu_holidays),
+        ),
+        2018: ((MAY, 21, special_in_lieu_holidays),),
+        2019: ((SEP, 30, special_in_lieu_holidays),),
+        2020: (
+            (MAY, 11, special_in_lieu_holidays),
+            (AUG, 17, sangkranta_in_lieu_covid),
+            (AUG, 18, sangkranta_in_lieu_covid),
+            (AUG, 19, sangkranta_in_lieu_covid),
+            (AUG, 20, sangkranta_in_lieu_covid),
+            (AUG, 21, sangkranta_in_lieu_covid),
+        ),
+    }
+    supported_languages = ("en_US", "km", "th")
+
+    def __init__(self, *args, **kwargs):
+        InternationalHolidays.__init__(self)
+        self.thls = _ThaiLunisolar()
```

Comment:
```suggestion
        self.thls = _ThaiLunisolar(KHMER_CALENDAR)
```
And will be no need to pass `KHMER_CALENDAR` parameter in `makha_bucha_date()` and `visakha_bucha_date()`

Reply:
Nice catch, fixed <3